sw/device/lib/dif/dif_csrng.c

```diff
@@ -226,57 +156,80 @@ dif_result_t dif_csrng_get_cmd_interface_status(
-                  kDifCsrngFifoCmd);
-    BITSET_INSERT(status->unhealthy_fifos, CSRNG_ERR_CODE_SFIFO_GENBITS_ERR_BIT,
-                  kDifCsrngFifoGenBits);
-    BITSET_INSERT(status->unhealthy_fifos, CSRNG_ERR_CODE_SFIFO_CMDREQ_ERR_BIT,
-                  kDifCsrngFifoCmdReq);
-    BITSET_INSERT(status->unhealthy_fifos, CSRNG_ERR_CODE_SFIFO_RCSTAGE_ERR_BIT,
-                  kDifCsrngFifoRcStage);
-    BITSET_INSERT(status->unhealthy_fifos, CSRNG_ERR_CODE_SFIFO_KEYVRC_ERR_BIT,
-                  kDifCsrngFifoKeyVrc);
-    BITSET_INSERT(status->unhealthy_fifos, CSRNG_ERR_CODE_SFIFO_UPDREQ_ERR_BIT,
-                  kDifCsrngFifoUpdateReq);
-    BITSET_INSERT(status->unhealthy_fifos, CSRNG_ERR_CODE_SFIFO_BENCREQ_ERR_BIT,
-                  kDifCsrngFifoBencRec);
-    BITSET_INSERT(status->unhealthy_fifos, CSRNG_ERR_CODE_SFIFO_BENCACK_ERR_BIT,
-                  kDifCsrngFifoBencAck);
-    BITSET_INSERT(status->unhealthy_fifos, CSRNG_ERR_CODE_SFIFO_PDATA_ERR_BIT,
-                  kDifCsrngFifoPData);
-    BITSET_INSERT(status->unhealthy_fifos, CSRNG_ERR_CODE_SFIFO_FINAL_ERR_BIT,
-                  kDifCsrngFifoFinal);
-    BITSET_INSERT(status->unhealthy_fifos,
-                  CSRNG_ERR_CODE_SFIFO_GBENCACK_ERR_BIT, kDifCsrngFifoGBencAck);
-    BITSET_INSERT(status->unhealthy_fifos,
-                  CSRNG_ERR_CODE_SFIFO_GRCSTAGE_ERR_BIT, kDifCsrngFifoGrcStage);
-    BITSET_INSERT(status->unhealthy_fifos, CSRNG_ERR_CODE_SFIFO_GGENREQ_ERR_BIT,
-                  kDifCsrngFifoGGenReq);
-    BITSET_INSERT(status->unhealthy_fifos,
-                  CSRNG_ERR_CODE_SFIFO_GADSTAGE_ERR_BIT, kDifCsrngFifoGadStage);
-    BITSET_INSERT(status->unhealthy_fifos, CSRNG_ERR_CODE_SFIFO_BLKENC_ERR_BIT,
-                  kDifCsrngFifoBlockEnc);
-
-    BITSET_INSERT(status->errors, CSRNG_ERR_CODE_CMD_STAGE_SM_ERR_BIT,
-                  kDifCsrngErrorCmdStageSm);
-    BITSET_INSERT(status->errors, CSRNG_ERR_CODE_MAIN_SM_ERR_BIT,
-                  kDifCsrngErrorMainSm);
-    BITSET_INSERT(status->errors, CSRNG_ERR_CODE_DRBG_GEN_SM_ERR_BIT,
-                  kDifCsrngErrorDrbgGenSm);
-    BITSET_INSERT(status->errors, CSRNG_ERR_CODE_DRBG_UPDBE_SM_ERR_BIT,
-                  kDifCsrngErrorDrbgUpdateBlockEncSm);
-    BITSET_INSERT(status->errors, CSRNG_ERR_CODE_DRBG_UPDOB_SM_ERR_BIT,
-                  kDifCsrngErrorDrbgUpdateOutBlockSm);
-    BITSET_INSERT(status->errors, CSRNG_ERR_CODE_AES_CIPHER_SM_ERR_BIT,
-                  kDifCsrngErrorAesSm);
-    BITSET_INSERT(status->errors, CSRNG_ERR_CODE_CMD_GEN_CNT_ERR_BIT,
-                  kDifCsrngErrorGenerateCmdCounter);
-    BITSET_INSERT(status->errors, CSRNG_ERR_CODE_FIFO_WRITE_ERR_BIT,
-                  kDifCsrngErrorFifoWrite);
-    BITSET_INSERT(status->errors, CSRNG_ERR_CODE_FIFO_READ_ERR_BIT,
-                  kDifCsrngErrorFifoRead);
-    BITSET_INSERT(status->errors, CSRNG_ERR_CODE_FIFO_STATE_ERR_BIT,
-                  kDifCsrngErrorFifoFullAndEmpty);
+    status->unhealthy_fifos =
```

Comment:
Isn't worth having a look up table mapping the `kDifCsrngFifo*` to `CSRNG_ERROR_CODE_SFIFO*` and iterate through them here?